### PR TITLE
*: Forbid add a partition with placement on a table which has tiflash replicas

### PR DESCRIPTION
**Pull request by lcwangchao**
### What problem does this PR solve?

Issue Number: ref #30741

Problem Summary:

In PR:  #30844 we had forbid many placement operations but forgot `add table partition` . This pr added this.  

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
Forbid add a partition with placement on a table which has tiflash replicas
```

### CODE REVIEW

**Review by hawkingrei — APPROVED** [review]

File: ddl/partition.go [hunk 1]
```diff
@@ -94,6 +94,10 @@ func (w *worker) onAddTablePartition(d *ddlCtx, t *meta.Meta, job *model.Job) (v
 		return ver, err
 	}
 
+	if tblInfo.TiFlashReplica != nil && tblInfo.TiFlashReplica.Count > 0 {
+		return 0, errors.Trace(ErrIncompatibleTiFlashAndPlacement)
```
Comment (xhebox) [inline ddl/partition.go:98]: Maybe return ver

**Review by xhebox — APPROVED** [review]

**Pull request merged**